pyproject.toml

```diff
@@ -0,0 +1,41 @@
+[tool.poetry]
+name = "npc-gzip"
+version = "0.1.0"
+description = "Code for Paper: “Low-Resource” Text Classification: A Parameter-Free Classification Method with Compressors"
+authors = [
+    "Gin Jiang <bazingagin@gmail.com>",
+    "Zach Bloss <zacharybloss@gmail.com>",
+]
+maintainers = [
+    "Zach Bloss <zacharybloss@gmail.com>",
+    "Gin Jiang <bazingagin@gmail.com>",
+]
+license = "MIT"
+readme = "README.md"
+homepage = "https://github.com/bazingagin/npc_gzip"
+repository = "https://github.com/bazingagin/npc_gzip"
+keywords = ["npc_gzip", "knn-gzip"]
+packages = [{include = "npc_gzip"}]
+
+[tool.poetry.dependencies]
+python = "^3.9"
+numpy = "^1.25.1"
+tqdm = "^4.65.0"
+
+[tool.poetry.group.test.dependencies]
+pytest = "^7.4.0"
+
+
+[tool.poetry.group.lint.dependencies]
+black = "^23.7.0"
+isort = "^5.12.0"
+
+
+[tool.poetry.group.examples.dependencies]
+torchtext = "^0.15.2"
+portalocker = "^2.7.0"
+scikit-learn = "^1.3.0"
+
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
```

Comment:
If we add this stanza to the end of `pyproject.toml`, then all three of our `pytest` commands can be simplified:

```toml
[tool.pytest.ini_options]
addopts = "--doctest-modules --ignore original_codebase/ --ignore examples/"
```

This would allow `--doctest-modules --ignore original_codebase/ --ignore examples/` to be removed from `publish-package.yml`, `test-package.yml`, and `README.md`. The main benefit would be to `README.md`, because the command for users to run just becomes `pytest` again.

In the future, modules in `examples` may gain tests, and then this might be changed. But I would consider that an advantage of this approach--the user will (probably) be able to keep doing the same simple thing, while benefiting from new functionality.

Reply:
Good idea, added!